### ISSUE TITLE
Description list extension generates incorrect HTML when using link reference definitions

### ISSUE THREAD

**Issue by ramsey** [thephpleague/commonmark#692]
**Version(s) affected**: 2.0.0

**Description**  
Link reference definitions appear to confuse the parser when using the description list extension.

For the given Markdown:

```markdown
[Lorem ipsum dolor sit amet][foo], consectetur adipiscing elit. Cras vitae
fringilla nulla. Etiam ac lectus scelerisque justo laoreet cursus. Mauris ut
dictum urna. Integer sit amet nibh aliquam, aliquet orci ut, condimentum diam.
Vestibulum varius purus at pulvinar feugiat. Donec facilisis mauris non sapien
ullamcorper semper. Nunc quis sapien eu metus tempor elementum a at mi. Fusce
at nisi et lectus lobortis rutrum.

[foo]: https://example.com
```

I expect the following HTML produced:

```html
<p><a href="https://example.com">Lorem ipsum dolor sit amet</a>, consectetur adipiscing elit. Cras vitae
fringilla nulla. Etiam ac lectus scelerisque justo laoreet cursus. Mauris ut
dictum urna. Integer sit amet nibh aliquam, aliquet orci ut, condimentum diam.
Vestibulum varius purus at pulvinar feugiat. Donec facilisis mauris non sapien
ullamcorper semper. Nunc quis sapien eu metus tempor elementum a at mi. Fusce
at nisi et lectus lobortis rutrum.</p>
```

However, the following HTML is actually produced:

```html
<dl>
<dt>[Lorem ipsum dolor sit amet][foo], consectetur adipiscing elit. Cras vitae</dt>
<dt>fringilla nulla. Etiam ac lectus scelerisque justo laoreet cursus. Mauris ut</dt>
<dt>dictum urna. Integer sit amet nibh aliquam, aliquet orci ut, condimentum diam.</dt>
<dt>Vestibulum varius purus at pulvinar feugiat. Donec facilisis mauris non sapien</dt>
<dt>ullamcorper semper. Nunc quis sapien eu metus tempor elementum a at mi. Fusce</dt>
<dt>at nisi et lectus lobortis rutrum.</dt>
<dd><p>https://example.com</p></dd>
</dl>
```

**How to reproduce**  

Run `composer require league/commonmark` and then execute the following script:

```php
<?php

declare(strict_types=1);

use League\CommonMark\Environment\Environment;
use League\CommonMark\Extension\CommonMark\CommonMarkCoreExtension;
use League\CommonMark\Extension\DescriptionList\DescriptionListExtension;
use League\CommonMark\MarkdownConverter;

require_once 'vendor/autoload.php';

$environment = new Environment();
$environment->addExtension(new CommonMarkCoreExtension());
$environment->addExtension(new DescriptionListExtension());

$converter = new MarkdownConverter($environment);

$content = <<<'MD'
[Lorem ipsum dolor sit amet][foo], consectetur adipiscing elit. Cras vitae
fringilla nulla. Etiam ac lectus scelerisque justo laoreet cursus. Mauris ut
dictum urna. Integer sit amet nibh aliquam, aliquet orci ut, condimentum diam.
Vestibulum varius purus at pulvinar feugiat. Donec facilisis mauris non sapien
ullamcorper semper. Nunc quis sapien eu metus tempor elementum a at mi. Fusce
at nisi et lectus lobortis rutrum.

[foo]: https://example.com

MD;

echo $converter->convertToHtml($content);
```

**Comment by colinodell** [comment]
It seems the code was looking for a semi-colon _anywhere_ in the line, not just at the start of the line :facepalm: Fixed via commit c701d5f.

**Comment by colinodell** [comment]
Fix has been released in 2.0.1! :tada:

**Comment by ramsey** [comment]
Thanks!
